var/spack/repos/builtin/packages/py-jupyterhub/package.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyJupyterhub(PythonPackage):
+    """Multi-user server for Jupyter notebooks."""
+
+    homepage = "https://pypi.org/project/jupyterhub"
+    url      = "https://pypi.io/packages/source/j/jupyterhub/jupyterhub-0.9.4.tar.gz"
+
+    version('0.9.4',    sha256='86b1cce446d4e8347e26913878858fc8964d103fde19b606fe37ccc5188d629d')
```

Comment:
maybe just add the latest version for now, it this is not yet tested (with all versions)